proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpHttpHandler.java

```diff
@@ -129,18 +139,26 @@ protected void handleHttpMessage(ChannelHandlerContext ctx, FullHttpRequest requ
             return;
           }
 
-          OtlpProtobufUtils.exportToWavefront(
-              otlpRequest, spanHandler, spanLogsHandler, preprocessorSupplier, spanLogsDisabled,
+          OtlpTraceUtils.exportToWavefront(
+              traceRequest, spanHandler, spanLogsHandler, preprocessorSupplier, spanLogsDisabled,
               spanSamplerAndCounter, defaultSource, discoveredHeartbeatMetrics, internalReporter,
               traceDerivedCustomTagKeys
           );
           break;
-
+        case "/v1/metrics/":
+          ExportMetricsServiceRequest metricRequest =
+              ExportMetricsServiceRequest.parseFrom(request.content().nioBuffer());
+          OtlpMetricsUtils.exportToWavefront(metricRequest,
+              metricsHandler, histogramHandler, preprocessorSupplier, defaultSource,
+              includeResourceAttrsForMetrics);
+          break;
         default:
-          String msg = "Unknown OTLP endpoint " + uri.getPath();
-          logWarning("WF-300: " + msg, null, ctx);
-          HttpResponse response = makeErrorResponse(Code.NOT_FOUND, msg);
-          writeHttpResponse(ctx, response, request);
+          /*
+          We use HTTP 200 for success and HTTP 400 for errors, mirroring what we found in
+          OTel Collector's OTLP Receiver code.
+         */
+          // TODO report internal metrics
```

Comment:
Has this `TODO` comment been addressed?

Reply:
Are there existing templates of internal metrics that other listeners report (e.g the listener on port 2878)? We don't want to invent new metrics that may not have value.

Reply:
These are the existing internal metrics we currently have for proxy: https://docs.wavefront.com/monitoring_proxies.html#monitoring-proxy-health-with-the-usage-dashboard. We also have internal metrics for errors in system/CollectorApiServer.

Reply:
Thanks for the info Joanna. Looking at other parts of the WF Proxy code, it appears that updating internal metrics upon sending 400 http status is not done. After consulting with my team, we feel that we don't want to add special internal metrics for 400 statuses just for OTEL.  If later we want to start an internal metric for 400 status codes, it should be a separate story and apply to all intake formats and not be a part of this PR.